### PR TITLE
Closes #48 - Initial commit for adding responses to inventory probes

### LINKED ISSUE #48: Distributed bus must respond to inventory queries on appropriate subjects
We need a way to get a list of the actors currently up and running in the lattice. The in-proc/default bus does **not** need to implement this feature.

My current thinking on this is to just have the distributed bus subscribe to the `wasmbus.control.actors` subject. A consumer wishing to obtain the lattice's actor list need simply place an empty request on that subject and the bus will answer.

## Technical Options

As usual with distributed systems, the hard part is in the tiny details.

* Option 1 - In this case, if we don't queue subscribe to this subject, then all bus instances (1 per wascc host) will respond to this query with the list of actors within that given host. It will then be up to the consumer/requester to aggregate that list. **Pros** - "distributed friendly", no shared state needs to be maintained by the bus instances, and they don't need to be aware of the state of other instances (e.g. we won't need to get into the distributed cache business). **Cons** - requires some post-processing on the part of the consume
* Option 2 - Queue subscribe to the subject so that only one bus instance will answer the request. In this scenario, each bus must be aware of the bus membership of all other bus instances within the lattice. tl;dr each bus becomes a node in a distributed cache. **Pros** - consumer needs to do no work to consume this data, it is already normalized. **Cons** - the lattice now needs to get in the business of managing an eventually consistent "ring" of membership data, potentially adding an enormous amount of complexity to a product designed to remove developer friction.

In either scenario, the bus should dispatch `OP_HEALTH_REQUEST`s to each registered actor so that it can reliably indicate which actors are _alive_ and which ones may have had their processing thread panic and are no longer responding to dispatches.

### PR DESCRIPTION
**Pull request by autodidaddict**

### CODE REVIEW

File: src/bus/lattice.rs [hunk 1]
```diff
@@ -77,6 +96,92 @@ impl DistributedBus {
...
 }
 
+fn spawn_inventory_handler(
+    nc: Arc<nats::Connection>,
+    host_id: String,
+    claims: Arc<RwLock<HashMap<String, Claims<Actor>>>>,
+    bindings: Arc<RwLock<BindingsList>>,
+    caps: Arc<RwLock<HashMap<RouteKey, CapabilityDescriptor>>>,
+) -> Result<()> {
+    let _ = nc.subscribe(INVENTORY_SUBJECT)?.with_handler(move |msg| {
```
Comment (brooksmtownsend) [inline src/bus/lattice.rs:106]: Is there a reason you're assigning the result of this `subscribe` method to nothing?

File: src/bus/lattice.rs [hunk 2]
```diff
@@ -77,6 +96,92 @@ impl DistributedBus {
...
+    msg.respond(serde_json::to_vec(&ir).unwrap())
+        .map_err(|e| e.into())
+}
+
+fn respond_with_caps(
+    msg: nats::Message,
+    host: String,
+    caps: Arc<RwLock<HashMap<RouteKey, CapabilityDescriptor>>>,
+) -> std::result::Result<(), std::io::Error> {
+    let mut capabilities = vec![];
```
Comment (brooksmtownsend) [inline src/bus/lattice.rs:170]: Here you use `vec![]`, above (line 147) you use Vec::<Binding>::new(), I know this is nitpicky but do we have a preferred method to instantiate a new empty vector?

File: src/bus/lattice.rs (hunk 1 above)
Comment (autodidaddict) [reply src/bus/lattice.rs:106]: Yep, this is a common pattern for signaling that we intend to invoke a result-returning function, but we don't care if it fails. There is a larger concern outside the scope of this PR where we need to address some of the patterns we're following for error handling in background threads.

File: src/bus/lattice.rs (hunk 2 above)
Comment (autodidaddict) [reply src/bus/lattice.rs:170]: I usually prefer `Vec::new()`. From the compiler's perspective, `vec![]` and `Vec::new()` are identical (the macro expands to the latter).

I think what we should ultimately be looking for is opportunities to use `Vec::with_capacity`. If we know the number of items we're going to have, or even a ballpark number, then we might be able to save some _time_ by pre-allocating in bulk.

**Pull request merged**